From a015aa6dfd02f794fed8a629dc1158acf6bb6521 Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Fri, 24 May 2013 17:38:45 +0700
Subject: Dalvik: Fix_Chaining_Cells pass should use kAllNodesAndNew iterator

BZ: 110737

Fix_Chaining_Cells pass creates a new basic blocks but uses kAllNodes iterator.
It results in assert in iterator which does not expect new basic blocks.
Patch changes the type of iterator to work correctly.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I3c48174d57301666634f9968395e0fd02b6b30ad
Orig-MCG-Change-Id: I7fb6413f9b41c025eb3573960d948063c08294d3
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/LoopOpt.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/LoopOpt.cpp b/vm/compiler/LoopOpt.cpp
index 71d8934..39ad51f 100644
--- a/vm/compiler/LoopOpt.cpp
+++ b/vm/compiler/LoopOpt.cpp
@@ -87,7 +87,7 @@ START_PASSES
     NEW_PASS ("Write_Back_Registers", kAllNodes, 0, 0, 0, 0, dvmCompilerWriteBackAll, 0, 0),
     NEW_PASS ("Registerization_ME", kAllNodes, 0, dvmCompilerRegisterizeGate,
                 0, dvmCompilerRegisterize, 0, 0, 0),
-    NEW_PASS ("Fix_Chaining_Cells", kAllNodes, 0, 0,
+    NEW_PASS ("Fix_Chaining_Cells", kAllNodesAndNew, 0, 0,
                 0, 0, dvmCompilerFixChainingCellPredecessors, 0, kOptimizationBasicBlockChange),
     NEW_PASS ("Insert_Prebackward_Block", kAllNodesAndNew, 0, 0,
                     0, dvmCompilerInsertPreBackward, 0, 0, kOptimizationBasicBlockChange),
-- 
1.7.4.1

